README.md

```diff
@@ -123,24 +33,117 @@ Run `make test` to run all the unit and integration tests.
+Iterators saves last processed value from `orderingColumn` column to position to field `SnapshotLastProcessedVal`. 
+If snapshot stops it will parse position from last record and will try gets row where `{{orderingColumn}} > {{position.SnapshotLastProcessedVal}}`
+
+
+When all records are returned, the connector switches to the CDC iterator.
+
+This behavior is enabled by default, but can be turned off by adding "snapshot":"false" to the Source configuration.
+
+### Change Data Capture (CDC)
+
+This connector implements CDC features for DB2 by adding a tracking table and triggers to populate it. The tracking
+table has the same name as a target table with the prefix `CONDUIT_TRACKING_`. The tracking table has all the
+same columns as the target table plus three additional columns:
+
+| name                            | description                                          |
+|---------------------------------|------------------------------------------------------|
+| `CONDUIT_TRACKING_ID`           | Autoincrement index for the position.                |
+| `CONDUIT_OPERATION_TYPE`        | Operation type: `insert`, `update`, or `delete`.     |
+| `CONDUIT_TRACKING_CREATED_DATE` | Date when the event was added to the tacking table.  |
+
+
+Triggers have name pattern `CONDUIT_TRIGGER_{{operation_type}}_{{table}}`. 
+
+
+Queries to retrieve change data from a tracking table are very similar to queries in a Snapshot iterator, but with 
+`CONDUIT_TRACKING_ID` ordering column.
+
+CDC iterator periodically clears rows which were successfully applied from tracking table. 
+It collects `CONDUIT_TRACKING_ID` inside the `Ack` method into a batch and clears the tracking table every 5 seconds.
+
+Iterator saves the last `CONDUIT_TRACKING_ID` to the position from the last successfully recorded row.
+
+If connector stops, it will parse position from the last record and will try 
+to get row where `{{CONDUIT_TRACKING_ID}}` > `{{position.CDCLastID}}`.
+
+
+### CDC FAQ
+
+#### Is it possible to add/remove/rename column to table?
+
+Yes. You have to stop the pipeline and do the same with conduit tracking table.
+For example:
+```sql
+ALTER TABLE CLIENTS
+ADD COLUMN phone VARCHAR(18);
+
+ALTER TABLE CONDUIT_TRACKING_CLIENTS
+    ADD COLUMN phone VARCHAR(18);
+```
+
+#### I accidentally removed tracking table.
+
+You have to restart pipeline, tracking table will be recreated by connector.
+
+#### I accidentally removed table.
+
+You have to stop the pipeline, remove the conduit tracking table, and then start the pipeline.
+
+#### Is it possible to change table name?
+
+Yes. Stop the pipeline, change the value of the `table` in the Source configuration, 
+change the name of the tracking table using a pattern `CONDUIT_TRACKING_{{TABLE}}`
+
+#### Is it possible to use two identical Db2 source connectors with the same configs on different pipelines?
+
+No. You can add more destination connectors on pipeline if you need it.
```

Comment:
Can we use a different tracking table and trigger name for each connector?

Reply:
We can. It will be looks like it: we will create names using current time  and we will save this pattern in `Position` to field  `NamePattern`. It is will be some problem. For example :
1. User starts pipeline with db2 source connector.
2. Connector checks that `position` parameter is nil for `Open` method and create tracking table and triggers with name `test210000`.
3. But `table` doesn't has any data and connector doesn't create record and doesn't save position.
4. User stop pipeline.
5. User start pipeline and connector  will create new additional tracking table and triggers  with name `test210200`. because doesn't see position. 

For this case we can add additional check inside `Teardown` method.  We checks if connector doesn't send any record we will remove current tracking table and triggers.

What do you think about it?

Reply:
yep. Done